src/contexts/CartContextProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import { ReactNode, createContext } from 'react';
+import { CartItem } from '../types/CartItem';
+import { useLocalStorage } from '../hooks/useLocalStorage';
+
+type Props = {
+  children: ReactNode,
+};
+
+type CartContextType = {
+  cart: CartItem[],
+  addToCart: (newProduct: CartItem) => void,
+  removeFromCart: (productId: string) => void,
+  quantityCart: (productId: string, action: Action) => void,
+};
+
+export const CartContext = createContext<CartContextType>({
+  cart: [],
+  addToCart: () => {},
+  removeFromCart: () => {},
+  quantityCart: () => {},
+});
+
+export enum Action {
+  Increase = 'increase',
+  Decrease = 'decrease',
+}
+
+export const CartContextProvider: React.FC<Props> = ({ children }) => {
+  const [cart, setCart] = useLocalStorage<CartItem[]>('cart', []);
+
+  const addToCart = (product: CartItem) => {
+    setCart([
+      ...cart,
+      product,
+    ]);
+  };
+
+  const quantityCart = (productId: string, action: Action) => {
```

Comment:
let's rename it - sounds like a variable that stores amount (quantity) of products in the cart